src/plugins/validate-semantic/index.js

```diff
@@ -110,3 +112,13 @@ function makeTraverseOnce(getSystem) {
     return deferred.promise.then( a => a[name] )
   }
 }
+
+function makeMemoizedResolveSubtree(system) {
+  const cacheKeymaker = (obj, path) => {
+    return `${obj.toString()} ${path.join("<>")}`
```

Comment:
is toString done on the whole object? It might be better to use the internal hash of the object, which should be unique?

Reply:
@ponelat yeah, it is done on the whole object - I didn't test `.hashCode()`, but a pr is welcome! 😄